src/bin/pg_autoctl/cli_perform.c

```diff
@@ -192,6 +192,20 @@ cli_perform_failover_getopts(int argc, char **argv)
 
 	cli_common_get_set_pgdata_or_exit(&(options.pgSetup));
 
+	if (!keeper_config_set_pathnames_from_pgdata(&(options.pathnames),
```

Comment:
would it make sense to move `keeper_config_set_pathnames_from_pgdata ` call into `cli_common_ensure_formation ` to make it robust for other cases?

Reply:
I though about this, but in some cases (see `cli_get_name_getopts`) it has been done already in a way that's not easy to refactor:

``` c
	/* now that we have the command line parameters, prepare the options */
	(void) prepare_keeper_options(&options);

	/* ensure --formation, or get it from the configuration file */
	if (!cli_common_ensure_formation(&options))
	{
		/* errors have already been logged */
		exit(EXIT_CODE_BAD_ARGS);
	}
```

The call to `prepare_keeper_options` is taking care of the pathnames, and is not something we want to do in every command either.